plugins/modules/packaging/os/copr.py

```diff
@@ -0,0 +1,477 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Silvie Chlupova <schlupov@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: copr
+short_description: Manage one of the Copr repositories
+version_added: 2.0.0
+description: This module can enable, disable or remove the specified repository.
+author: Silvie Chlupova (@schlupov) <schlupov@redhat.com>
+requirements:
+    - dnf
+    - dnf-plugins-core
+options:
+    host:
+        description: The Copr host to work with.
+        default: copr.fedorainfracloud.org
+        type: str
+    protocol:
+        description: This indicate which protocol to use with the host.
+        default: https
+        type: str
+    name:
+        description: Copr directory name, for example C(@copr/copr-dev).
+        required: true
+        type: str
+    state:
+        description:
+            - Whether to set this project as C(enabled), C(disabled) or C(absent).
+        default: enabled
+        type: str
+        choices: [absent, enabled, disabled]
+    chroot:
+        description:
+            - The name of the chroot that you want to enable/disable/remove in the project,
+              for example C(epel-7-x86_64). Default chroot is determined by the operating system,
+              version of the operating system, and architecture on which the module is run.
+        type: str
+"""
+
+EXAMPLES = r"""
+- name: Enable project Test of the user schlupov
+  community.general.copr:
+    host: copr.fedorainfracloud.org
+    state: enabled
+    name: schlupov/Test
+    chroot: fedora-31-x86_64
+
+- name: Remove project integration_tests of the group copr
+  community.general.copr:
+    state: absent
+    name: '@copr/integration_tests'
+"""
+
+RETURN = r"""
+repo_filename:
+  description: The name of the repo file in which the copr project information is stored
+  returned: success
+  type: str
+  sample: _copr:copr.fedorainfracloud.org:group_copr:integration_tests.repo
+
+repo:
+  description: Path to the project on the host
+  returned: success
+  type: str
+  sample: copr.fedorainfracloud.org/group_copr/integration_tests
+"""
+
+import stat
+import os
+from urllib import error
+from pathlib import Path
+import dnf
+import dnf.cli
+import dnf.repodict
+from dnf.conf import Conf
+from dnfpluginscore import _, logger
```

Comment:
```suggestion
import traceback

try:
    import dnf
    import dnf.cli
    import dnf.repodict
    from dnf.conf import Conf
    from dnfpluginscore import _, logger
    HAS_DNF_PACKAGES = True
except ImportError:
    DNF_IMP_ERR = traceback.format_exc()
    HAS_DNF_PACKAGES = False

from ansible.module_utils.basic import missing_required_lib  
```
and then in the module:
```.py
    if not HAS_DNF_PACKAGES:
        module.fail_json(msg=missing_required_lib("dnf"), exception=DNF_IMP_ERR)
```
(Or whatever the Python package to install is called.)